community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java

```diff
@@ -3866,13 +3891,95 @@ public void fileMappedWithDeleteOnCloseMustNotExistAfterLastUnmap() throws Excep
         File file = file( "a" );
         try ( PagedFile ignore = pageCache.map( file, filePageSize ) )
         {
-            pageCache.map( file, filePageSize, StandardOpenOption.DELETE_ON_CLOSE ).close();
+            pageCache.map( file, filePageSize, DELETE_ON_CLOSE ).close();
         }
         expectedException.expect( NoSuchFileException.class );
         pageCache.map( file, filePageSize );
     }
 
     @Test
+    public void fileMappedWithDeleteOnCloseShouldNotFlushDirtyPagesOnClose() throws Exception
+    {
+        AtomicInteger flushCounter = new AtomicInteger();
+        PageSwapperFactory swapperFactory = flushCountingPageSwapperFactory( flushCounter );
+        swapperFactory.setFileSystemAbstraction( fs );
+        PageCache cache = createPageCache( swapperFactory, maxPages, pageCachePageSize, PageCacheTracer.NULL,
+                PageCursorTracerSupplier.NULL );
+        File file = file( "a" );
+        try ( PagedFile pf = cache.map( file, filePageSize, DELETE_ON_CLOSE );
+              WritableByteChannel channel = pf.openWritableByteChannel() )
+        {
+            generateFileWithRecords( channel, recordCount, recordSize );
+        }
+        assertThat( flushCounter.get(), lessThan( recordCount / recordsPerFilePage ) );
+    }
+
+    @Test
+    public void mustFlushAllDirtyPagesWhenClosingPagedFileThatIsNotMappedWithDeleteOnClose() throws Exception
```

Comment:
Is this test not testing the same thing as `dirtyPagesMustBeFlushedWhenThePagedFileIsClosed` (also added in this PR)